src/commands/deleteVirtualMachine/deleteVirtualMachine.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { ComputeManagementClient } from "@azure/arm-compute";
+import { DialogResponses, IActionContext, IAzureQuickPickItem } from "vscode-azureextensionui";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../localize";
+import { VirtualMachineTreeItem } from "../../tree/VirtualMachineTreeItem";
+import { createComputeClient } from "../../utils/azureClients";
+import { getNameFromId } from "../../utils/azureUtils";
+import { nonNullProp, nonNullValue } from "../../utils/nonNull";
+import { IDeleteChildImplContext, ResourceToDelete, virtualMachineLabel } from "./deleteConstants";
+import { getResourcesAssociatedToVm } from "./getResourcesAssociatedToVm";
+import { promptResourcesToDelete } from "./promptResourcesToDelete";
+
+export async function deleteVirtualMachine(context: IActionContext & Partial<IDeleteChildImplContext>, node?: VirtualMachineTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<VirtualMachineTreeItem>(VirtualMachineTreeItem.allOSContextValue, context);
+    }
+
+    // if we can't retrieve the disk name, it's highly likely that it's the same as the vmName if it was created from the extensionf
+    const diskName: string = node.data.storageProfile?.osDisk?.managedDisk?.id ? getNameFromId(node.data.storageProfile.osDisk.managedDisk.id) : node.name;
```

Comment:
So the only reason I didn't do that is because I wanted to account for the case where we don't have a VM to get the resources from.

Do you think it'd be good to pass in the delete context and if it has a context.vm property, then get the diskName/networkName, but otherwise, the context could have it set prior to calling deleteVirtualMachine?

Reply:
Why wouldn't you have a VM?

Reply:
This is the scenario where VM creation fails.  Since when we create, we have to create literally everything before the VM, we've had situations where the user has every component except the VM itself.  This code would be useful to roll everything back and clean up all the resources.

Reply:
Oh...that's not a part of this PR, yet, right?

Reply:
No, but I didn't want to make the vm a required parameter when I had this in mind. 🤔 I guess I can just implement my idea after the fact though

Reply:
Yeah I don't really know what to say because I can't see the code, but couldn't vm just be optional instead of required?